### ISSUE TITLE
Add Object Detection Models

### ISSUE THREAD

**Issue by ztsv-av** [ztsv-av/general_model#11]
We only have classification networks. For the upcoming competition, we need to implement object detection networks, which I specified in models.py (commented). We can't import them from tf.keras, so we should try to find and download pretrained models from other resources. 

![image](https://user-images.githubusercontent.com/73081144/121988153-10a48b80-cd57-11eb-8acf-1d17d160a06a.png)

PP-YOLO is the newest (Jul, 2020) version of YOLO algorithm, which is why its performance is not included in the screenshot above.

**Comment by powerswing** [comment]
Keras doesn't have any object detection models in applications. Tensorflow has only four models (see [**here**](https://github.com/tensorflow/models/tree/master/official#object-detection-and-segmentation)).

It looks like writing those models by yourself is super long. Look for developed solutions (like [**this YOLOv3**](https://github.com/experiencor/keras-yolo3) and a [**tutorial**](https://towardsdatascience.com/yolo-v3-object-detection-with-keras-461d2cfccef6) to it) but I'm not sure if they are understandable enough.

Try those which are the fastest to implement. For the time's sake, run those models separately without embedding them into the class we're building.

>I would run those models on the preprocessed data, each model separately using their guidelines, and save the weights

**Comment by powerswing** [comment]
For reprocessing, look for quality information - this has to be done only once but done properly.

I think now the only preprocessing you have to do is rescaling (say, to 512x512) and centering, same applies to images with boxes.

Adding contrast and removing noise (as discussed [**here**](https://arxiv.org/pdf/1808.10858.pdf)) may be passed into permutations (so each batch/image is populated with different noise/contrast levels). What is universal, though, is that all the images are normalized using mean and standard deviation after permutations.

**Comment by powerswing** [comment]
I think you'll lose time if you start with box models first. I can try to look deeper into those framework I mentioned above during this weekend.

> Keras doesn't have any object detection models in applications. Tensorflow has only four models (see [**here**](https://github.com/tensorflow/models/tree/master/official#object-detection-and-segmentation)).
> 
> It looks like writing those models by yourself is super long. Look for developed solutions (like [**this YOLOv3**](https://github.com/experiencor/keras-yolo3) and a [**tutorial**](https://towardsdatascience.com/yolo-v3-object-detection-with-keras-461d2cfccef6) to it) but I'm not sure if they are understandable enough.
> 
> Try those which are the fastest to implement. For the time's sake, run those models separately without embedding them into the class we're building.
> 
> > I would run those models on the preprocessed data, each model separately using their guidelines, and save the weights
> 
> But really, you should finish preprocessing first, because this will allow you to train classification models right away (by nights) and work on box models by day.
